python/paddle/tensor/random.py

```diff
@@ -17,6 +17,108 @@
+            output Tensor. Supported data types: int64, int32. Default: int32.
+        device (str, optional): Specific the output variable to be saved in cpu
+            or gpu memory. Supported None, 'cpu', 'gpu'. If it is None, the output
+            variable will be automatically assigned devices.
+            Default: False.
+        stop_gradient (bool, optional): Whether grad should record operations 
+            on the returned tensor. Default: True.
+        seed (int, optional): Random seed used for permute samples. If seed is 
+            equal to 0, it means use a seed generated by the system. Note that 
+            if seed is not 0, this operator will always generate the same random 
+            permutation every time. Default: 0.
+
+    Returns:
+        Variable: A Tensor with a random permutation of integers from 0 to n - 1,
+            and the shape of returned tensor is [n].
+
+    Raise:
+        TypeError: The dtype must be one of int32 and int64, and the data type 
+            of out Tensor must be the same as the dtype. 
+        ValueError: The input n should be greater than 0 in randperm op.
+        ValueError: The input device should in [None, 'cpu', 'gpu'].
+
+    Examples:
+        .. code-block:: python
+	    import paddle
+	    import paddle.fluid as fluid
+
+	    num = 6
+	    is_use_gpu = False
+
+	    data_1 = paddle.tensor.randperm(num)
+	    fluid.layers.Print(data_1)
+
+	    data_2 = paddle.tensor.randperm(num, dtype="int32", seed=1)
+	    fluid.layers.Print(data_2)
+
+	    data_3 = paddle.tensor.randperm(num, stop_gradient=False, device="cpu")
+	    fluid.layers.Print(data_3)
+
+	    paddle.tensor.randperm(num, out=data_3)
+	    fluid.layers.Print(data_3)
+
+	    place = fluid.CUDAPlace(0) if is_use_gpu else fluid.CPUPlace()
+	    exe = fluid.Executor(place)
+	    exe.run(fluid.default_startup_program())
+	    exe.run()
+ 
+    """
+
+    if n < 1:
+        raise ValueError("The input n should be greater than 0 in randperm op.")
+    check_dtype(dtype, 'dtype', ['int64', 'int32'], 'randperm')
+    dtype = convert_dtype(dtype)
+    if device not in [None, 'cpu', 'gpu']:
+        raise ValueError("The input device should in [None, 'cpu', 'gpu'].")
+    check_type(stop_gradient, 'stop_gradient', bool, 'randperm')
+
+    helper = LayerHelper("randperm", **locals())
+    if out is None:
+        out = helper.create_variable_for_type_inference(dtype=dtype)
+    else:
+        check_variable_and_dtype(out, 'out', [dtype], 'randperm')
+    if stop_gradient:
+        out.stop_gradient = True
+
+    inputs = dict()
+    outputs = {'Out': [out]}
+    attrs = {'n': n, 'dtype': out.dtype, 'seed': seed}
+
+    if device is None:
+        helper.append_op(
+            type='randperm', inputs=inputs, outputs=outputs, attrs=attrs)
+    else:
+        with device_guard(device):
+            helper.append_op(
+                type='randperm', inputs=inputs, outputs=outputs, attrs=attrs)
```

Comment:
The attr(device) supports None in device_guard API, so if statement can be removed.

Reply:
> The attr(device) supports None in device_guard API, so if statement can be removed.

Done. TKS